das/stats.go

```diff
@@ -0,0 +1,30 @@
+package das
+
+// SamplingStats collects information about the DASer process. Currently, there are
+// only two sampling routines: the main sampling routine which performs sampling
+// over current network headers, and the `catchUp` routine which performs sampling
+// over past headers from the last sampled checkpoint.
+type SamplingStats struct {
+	// all headers before SampledBefore were successfully sampled
+	SampledBefore uint64 `json:"sampled_before_height"`
+	// MaxKnown is the height of the newest known header
+	MaxKnown uint64 `json:"max_known_height"`
+	// Failed contains all skipped header's heights with corresponding try count
+	Failed map[uint64]int `json:"failed,omitempty"`
```

Comment:
Some headers might have failed to be retrieved from header store (`GetByHeight` error), so it will require to store 2 Failed maps for retry:
   - failed to obtain from store  - stored by height
   - failed to be sampled - store by hash
   And it will make much harder retry logic.
Adding handling by hash would make coordinator much more complicated, right now it tracks all planned / in progress / failed headers by height. Internal hashes cache could do the job, but I thought that would be an overkill for now.

Reply:
I liked the idea of optimizing IO, but then I got a question: do we need to optimize IO for the failed cases here? Also, considering the complexity it introduces that Vlad described